src/rgw/driver/rados/rgw_rados.h

```diff
@@ -853,7 +853,11 @@ class RGWRados
         std::list<rgw_obj_index_key> *remove_objs;
         ceph::real_time expiration_time;
         ceph::real_time unmod_since;
+        ceph::real_time last_mod_time_match;
         ceph::real_time mtime; /* for setting delete marker mtime */
+        uint64_t size_match = 0;
```

Comment:
0 is a valid value for If-Match-Size. if the client sends that, we won't enforce it in delete_obj():
```c++
  if (params.size_match != 0 && state->size != params.size_match) {
      return -ERR_PRECONDITION_FAILED;
```
lets use `std::optional<uint64_t>` for this so we can differentiate between 0 and the empty state